blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -22,21 +22,21 @@
 import com.rackspacecloud.blueflood.types.Locator;
 import com.rackspacecloud.blueflood.types.Metric;
 import com.rackspacecloud.blueflood.utils.TimeValue;
-import org.apache.commons.lang.StringUtils;
 import org.codehaus.jackson.annotate.JsonIgnore;
 import org.joda.time.DateTime;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import static com.rackspacecloud.blueflood.inputs.formats.MetricValidator.*;
+
 public class JSONMetricsContainer {
+
     private final String tenantId;
     private final List<JSONMetric> jsonMetrics;
     private List<Metric> delayedMetrics;
     private static final long delayedMetricsMillis = Configuration.getInstance().getLongProperty(CoreConfig.DELAYED_METRICS_MILLIS);
-    private static final long pastDiff = Configuration.getInstance().getLongProperty( CoreConfig.BEFORE_CURRENT_COLLECTIONTIME_MS );
-    private static final long futureDiff = Configuration.getInstance().getLongProperty( CoreConfig.AFTER_CURRENT_COLLECTIONTIME_MS );
 
     public JSONMetricsContainer(String tenantId, List<JSONMetric> metrics) {
```

Comment:
@ChandraAddala @shintasmith 
Also, I kinda like shinta idea of parsing the json as the container is created, but instead of having invalidMetrics we just have an error message list that can be return for getValidationErrors().  I refactored it just so and made it so that the validation is done only once per metric (when the container is created and as each metric is processed).

Check it out!